x/archivedb/key_value_reader.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package archivedb
+
+import (
+	"bytes"
+
+	"github.com/ava-labs/avalanchego/database"
+)
+
+type dbHeightReader struct {
+	db                 *archiveDB
+	height             uint64
+	heightLastFoundKey uint64
+}
+
+// Builds a database.Iterator where the next value is the requested *value*. If
+// the value is not found a database.ErrNotFound is returned.
+//
+// This is a private function which helps Get() and Has() to share the key
+// lookup logic
```

Comment:
Can we add a comment that the caller must call `iterator.Release()` even if this function returns an error?